### PR TITLE
📦  FEAT: [bulk/main] check csv file exists and created date

### PR DESCRIPTION
**Pull request by chancesmith**
## overall purpose
Tell dev if csv doesn't exist and if the file might be too old to build from.

## changes made
- check file exists
- check file created at date is less than a day

## screenshot
Screen Shot 2022-12-29 at 10 11 51 AM

### CODE REVIEW

File: csv-bulk-signature/src/main.ts [hunk 1]
```diff
@@ -28,7 +28,8 @@ interface TemplateData {
  * GLOBALS
  */
 const CSV_FILE = "./src/contacts.csv";
-const TEMPLATE = "./src/email-sig-template.html";
+const CSV_EXPIRATION_DAYS = 1;
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:31]: add days experation const

File: csv-bulk-signature/src/main.ts [hunk 2]
```diff
@@ -54,7 +55,7 @@ async function setupFolders() {
 }
 
 async function setupTemplate() {
-  const templateFile = await fs.promises.readFile(TEMPLATE, "utf8");
+  const templateFile = await fs.promises.readFile(TEMPLATE_FILE, "utf8");
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:58]: match naming from csv_file

File: csv-bulk-signature/src/main.ts [hunk 3]
```diff
@@ -209,13 +210,47 @@ async function filterCompleteContacts(contacts: Contact[]) {
...
+  return await fs.promises
+    .access(CSV_FILE)
+    .then(() => {
+      return true;
+    })
+    .catch(() => {
+      console.log("🙅‍♂️ contacts.csv file does not exist");
+      return false;
+    });
+}
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts]: check if file exists

File: csv-bulk-signature/src/main.ts [hunk 4]
```diff
@@ -209,13 +210,47 @@ async function filterCompleteContacts(contacts: Contact[]) {
...
+  if (diffDays > CSV_EXPERATION_DAYS) {
+    console.log(
+      `🙅‍♂️ CSV contacts file is older than ${CSV_EXPERATION_DAYS} day(s). Please update the file.`
+    );
+  } else {
+    console.log(
+      `👍 CSV contacts file is less than ${CSV_EXPERATION_DAYS} day(s) old`
+    );
+  }
+}
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts]: check file is older than expiration const

File: csv-bulk-signature/src/main.ts [hunk 5]
```diff
@@ -209,13 +210,47 @@ async function filterCompleteContacts(contacts: Contact[]) {
...
+
 /*
  * MAIN
  */
 async function main() {
   await setupFolders();
   const template = await setupTemplate();
 
+  if (!(await hasContactsFile())) return;
+  await checkContactsFileCreatedAt();
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:251]: apply new file checks
- stop script if file doesn't exist

File: csv-bulk-signature/src/main.ts [hunk 6]
```diff
@@ -209,13 +210,46 @@ async function filterCompleteContacts(contacts: Contact[]) {
...
+  return await fs.promises
+    .access(CSV_FILE)
+    .then(() => {
+      return true;
+    })
+    .catch(() => {
+      console.log("🙅‍♂️ contacts.csv file does not exist");
+      return false;
+    });
+}
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:223]: check if file exists

File: csv-bulk-signature/src/main.ts [hunk 7]
```diff
@@ -209,13 +210,46 @@ async function filterCompleteContacts(contacts: Contact[]) {
...
+  if (diffDays > CSV_EXPIRATION_DAYS) {
+    console.log(
+      `🙅‍♂️ CSV contacts file is older than ${CSV_EXPIRATION_DAYS} day(s). Please update the file.`
+    );
+  } else {
+    console.log(
+      `👍 CSV contacts file is less than ${CSV_EXPIRATION_DAYS} day(s) old`
+    );
+  }
+}
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:241]: check expiration of contacts.csv

**Review by adamnitti — APPROVED** [review]

**Pull request merged**